source/core/types/ut_executable.tpb

```diff
@@ -59,23 +60,27 @@ create or replace type body ut_executable is
     return ut_metadata.form_name(owner_name, object_name, procedure_name);
   end;
 
-  member procedure do_execute(self in out nocopy ut_executable, a_item in out nocopy ut_suite_item, a_listener in out nocopy ut_event_listener_base) is
+  member procedure do_execute(self in out nocopy ut_executable, a_item in out nocopy ut_suite_item, a_listener in out nocopy ut_event_listener_base,
```

Comment:
I'm thinking more and more if that functionality (checking for exceptions) should be part of the generic ut_executable, or should we build a separate `ut_executable_test` to clearly separate additional functionality that is dedicated for tests only, not for before/after blocks.
So the `ut_executable_test` could call the base ut_executable.do_execute, and then invoke the custom logic to check fo rexceptions.
In suite builder, we would construct `ut_executable_test` only for the TEST block, and for all the others, we would still use ut_executable

Reply:
I like this approach. I think it's better create a ut_executable_test class